src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -37,11 +37,17 @@ public function __construct(EntityManager $entityManager, ClassMetadata $class)
 
     public function createListQueryBuilder(string $locale, $taxonId = null): QueryBuilder
     {
-        $queryBuilder = $this->createQueryBuilder('o')
-            ->addSelect('translation')
-            ->innerJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
-            ->setParameter('locale', $locale)
-        ;
+        $queryBuilder = $this->createQueryBuilder('o');
```

Comment:
How about using the association hydrator for the translations?

The only downside that I'm thinking right now of is loading all translations for the current page items thus a greater memory usage.

Reply:
If a store has many locales (like 30-40) it would consume a lot of memory. I'm not a big fan of this proposition.

Reply:
It may be a tradeoff we are willing to pay - as the current solution does not work perfectly, nor do I see a direct path how to fix it. If someone would try it and the current tests are green, I'm open to doing another round of performance tests